src/rpc/claimtrie.cpp

```diff
@@ -22,143 +24,177 @@ uint160 ParseClaimtrieId(const UniValue& v, const std::string& strName)
     return result;
 }
 
+static CBlockIndex* BlockHashIndex(const uint256& blockHash)
+{
+    AssertLockHeld(cs_main);
+
+    if (mapBlockIndex.count(blockHash) == 0)
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Block not found");
+
+    CBlockIndex* pblockIndex = mapBlockIndex[blockHash];
+    if (!chainActive.Contains(pblockIndex))
+        throw JSONRPCError(RPC_INTERNAL_ERROR, "Block not in main chain");
+
+    if (chainActive.Tip()->nHeight > (pblockIndex->nHeight + MAX_RPC_BLOCK_DECREMENTS))
+        throw JSONRPCError(RPC_INTERNAL_ERROR, "Block too deep to regenerate it");
+
```

Comment:
Instead of "Block too deep to regenerate it" , how about just "Block is too deep"

Reply:
getnameproof says "Block too deep to generate proof"